safrole/safrole.asn

```diff
@@ -1,33 +1,57 @@
--- Syntax for Safrole protocol test vectors
+-- Safrole STF test vectors schema
 
 SafroleModule DEFINITIONS ::= BEGIN
 
--- Define constants
-IMPORTS validators-count, epoch-length FROM SafroleConstants;
+IMPORTS
+    TimeSlot, ValidatorsData, Entropy, EntropyBuffer,
+    TicketBody, TicketsAccumulator, TicketsOrKeys, TicketsExtrinsic,
+    EpochMark, TicketsMark, Ed25519Public, BandersnatchRingCommitment
+        FROM JamTypes;
 
--- Define basic types
-U8 ::= INTEGER (0..255)
-U32 ::= INTEGER (0..4294967295)
 
--- Define fixed-length arrays
-ByteArray32 ::= SEQUENCE (SIZE(32)) OF U8
-OpaqueHash ::= ByteArray32
-Ed25519Key ::= ByteArray32
-BlsKey ::= SEQUENCE (SIZE(144)) OF U8
-BandersnatchKey ::= ByteArray32
-EpochKeys ::= SEQUENCE (SIZE(epoch-length)) OF BandersnatchKey
-TicketsBodies ::= SEQUENCE (SIZE(epoch-length)) OF TicketBody
+State ::= SEQUENCE {
+    -- [τ] Most recent block's timeslot.
+    tau TimeSlot,
+    -- [η] Entropy accumulator and epochal randomness.
+    eta EntropyBuffer,
+    -- [λ] Validator keys and metadata which were active in the prior epoch.
+    lambda ValidatorsData,
+    -- [κ] Validator keys and metadata currently active.
+    kappa ValidatorsData,
+    -- [γ_k] Validator keys for the following epoch.
+    gamma-k ValidatorsData,
+    -- [ι] Validator keys and metadata to be drawn from next.
+    iota ValidatorsData,
+    -- [γ_a] Sealing-key contest ticket accumulator.
+    gamma-a TicketsAccumulator,
+    -- [γ_s] Sealing-key series of the current epoch.
+    gamma-s TicketsOrKeys,
+    -- [γ_z] Bandersnatch ring commitment.
+    gamma-z BandersnatchRingCommitment
+    -- [ψ_o'] Posterior offenders sequence.
+    post-offenders SEQUENCE OF Ed25519Public
```

Comment:
Why is there a posterior element in the state?

Reply:
I have received feedback in the chat suggesting that the input should include only elements found within the block, such as header items and extrinsics. The pre-state consists of elements already present in the state, which may or may not have been altered (e.g., prior, dagger, double dagger, posterior). All in all I found that it makes sense

Reply:
Nice catch, I'll fix it. Ty

Reply:
@ggwpez should be fixed